base/compiler/ssair/inlining.jl

```diff
@@ -46,6 +63,11 @@ struct ConstantCase
     ConstantCase(val) = new(val)
 end
 
+struct SomeCase
+    val::Any
+    SomeCase(val) = new(val)
```

Comment:
isn't this constructor useless?

Reply:
There is the module-wide `@nospecialize` annotation, and so this constructor is equivalent to `SomeCase(@nospecialize val) = new(val)`. Maybe the equivalent constructor would be inserted without this though.